core/spec/models/spree/metafield_spec.rb

```
@@ -0,0 +1,27 @@
+    it 'validates the type must match the metafield definition' do
+      metafield_definition = create(:metafield_definition, metafield_type: 'Spree::Metafields::ShortText')
+      metafield = build(:metafield, metafield_definition: metafield_definition, type: 'Spree::Metafields::LongText')
+      expect(metafield.valid?).to be false
+    end
+  end
+
+  context 'Scopes' do
+    it 'returns the metafields with the given key' do
+      metafield_definition = create(:metafield_definition, namespace: 'custom', key: 'foo')
+      metafield = create(:metafield, metafield_definition: metafield_definition)
+      expect(Spree::Metafield.with_key('custom', 'foo')).to include(metafield)
```

RuboCop (1.80.2)

[convention] Use `described_class` instead of `Spree::Metafield`.

(RSpec/DescribedClass)